### PR TITLE
fix activation conditions

### PR DESCRIPTION
**Pull request by beetlecrunch**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 1]
```diff
@@ -736,6 +744,29 @@ final case class AgreementApiServiceImpl(
     } yield updated
   }
 
+  private def extractArchivables(
+    activatingAgreementId: UUID,
+    agreements: Seq[AgreementManagement.Agreement]
+  ): Seq[AgreementManagement.Agreement] =
+    agreements.filter(a => ARCHIVABLE_STATES.contains(a.state) && a.id == activatingAgreementId)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]:
```suggestion
    agreements.filter(a => ARCHIVABLE_STATES.contains(a.state) && a.id != activatingAgreementId)
```
?

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 2]
```diff
@@ -722,8 +722,16 @@ final case class AgreementApiServiceImpl(
...
+      seed               <- getUpdateSeed()
+      existingAgreements <- agreementManagementService.getAgreements(
+        consumerId = Some(agreement.consumerId.toString),
+        eserviceId = Some(agreement.eserviceId.toString)
+      )
+      archivables = extractArchivables(agreement.id, existingAgreements)
       updated <- agreementManagementService.updateAgreement(agreement.id, seed)
+      _       <-
+        if (archivables.isEmpty) Future.unit
+        else Future.traverse(archivables)(archive).map(_ => ())
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: isn't this the same?
```suggestion
        Future.traverse(archivables)(archive).map(_ => ())
```

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala (hunk 1 above)
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: doesn't this extract just the agreement corresponding to `activatingAgreementId` ?

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 3]
```diff
@@ -996,28 +1027,14 @@ final case class AgreementApiServiceImpl(
...
-  def verifyFirstActivationConflictingAgreements(
-    agreement: AgreementManagement.Agreement
-  )(implicit contexts: Seq[(String, String)]): Future[Unit] = {
-    val conflictingStates: List[AgreementManagement.AgreementState] =
-      List(AgreementManagement.AgreementState.ACTIVE, AgreementManagement.AgreementState.SUSPENDED)
-    verifyConflictingAgreements(agreement, conflictingStates)
-  }
+    Future
+      .failed(AgreementNotInExpectedState(agreement.id.toString, agreement.state))
+      .unlessA(admittedStates.contains(agreement.state))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: do we prefer avoiding cats Future helpers?

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala (hunk 1 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: You're right.
Thanks!

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: Typo!

**Review by TonioGela — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**